### PR TITLE
[12.0][ADD] Add Maintenance Request Repair

### PR DESCRIPTION
**Pull request by marcelsavegnago**
Add module maintenance_request_repair

This is a bridge module between Maintenance and Repair"

### CODE REVIEW

**Comment by etobella** [comment]
Can you check Travis errors? @marcelsavegnago

**Comment by etobella** [comment]
Well, a few comments:

- If you are linking from MR to repairs, you should need to make a link also on repairs to MRs, just in order to navigate properly
- Why are you making this link? Repairs are for repairing of goods supplied by the company, and Maintenance is for your actives, so I don't understand it. Can you provide more information or a Use Case please?

**Comment by marcelsavegnago** [comment]
> Well, a few comments:
> 
> * If you are linking from MR to repairs, you should need to make a link also on repairs to MRs, just in order to navigate properly

Ok

> * Why are you making this link? Repairs are for repairing of goods supplied by the company, and Maintenance is for your actives, so I don't understand it. Can you provide more information or a Use Case please?

Yes, I understand the difference between the modules and with this link it is possible to manage internal repair orders and with that we can relate the parts and operations to provide the repair.

In addition, I am working with the fieldservice_maintenance module that allows you to manage external maintenance schedules and consequently it may be necessary to generate repair orders for the equipment that has had problems and.

https://github.com/OCA/field-service/tree/12.0/fieldservice_maintenance

File: maintenance_request_repair/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    'author': 'Escodoo,Odoo Community Association (OCA)',
+    'website': 'https://github.com/OCA/maintenance',
+    'depends': [
+        'maintenance',
+        'repair',
+    ],
+    'data': [
+        'views/maintenance_request.xml',
+    ],
+    'demo': [
```
Comment (sergiocorato) [inline maintenance_request_repair/__manifest__.py]: Please remove unused key

File: maintenance_request_repair/readme/DESCRIPTION.rst [hunk 2]
```diff
@@ -0,0 +1,7 @@
+This is a bridge module between Maintenance and Repair
+
+This module contains some new features for Maintenance modules.
+
+Maintenance
```
Comment (sergiocorato) [inline maintenance_request_repair/readme/DESCRIPTION.rst]:
```suggestion
Maintenance Request
```

**Review by sergiocorato — CHANGES_REQUESTED** [review]

File: maintenance_request_repair/readme/DESCRIPTION.rst (hunk 2 above)
Comment (marcelsavegnago) [reply maintenance_request_repair/readme/DESCRIPTION.rst]: Done

File: maintenance_request_repair/__manifest__.py (hunk 1 above)
Comment (marcelsavegnago) [reply maintenance_request_repair/__manifest__.py]: Done

**Comment by etobella** [comment]
@marcelsavegnago Link from repairs to maintenance request is missing. Can you review?

**Comment by etobella** [comment]
Now the field is created, but the link button is still missing :wink:

**Comment by marcelsavegnago** [comment]
@etobella I added the link button. Thanks if you can review it.

File: maintenance_request_repair/views/repair_order.xml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+            <field name="inherit_id" ref="repair.view_repair_order_form"/>
+            <field name="priority" eval="8"/>
+            <field name="arch" type="xml">
+                <div name="button_box" position="inside">
+                    <field name="maintenance_request_ids" invisible="1"/>
+                    <button type="object"
+                            name="action_view_maintenance"
+                            class="oe_stat_button"
+                            icon="fa-wrench"
+                            attrs="{'invisible': [('maintenance_request_count', '=', 0)]}"
```
Comment (etobella) [inline maintenance_request_repair/views/repair_order.xml]: Why invisible if it is 0? Does make sense to create a maintenance request from a repair order?

File: maintenance_request_repair/models/repair_order.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    def _compute_maintenance_request_count(self):
+        for repair in self:
+            repair.maintenance_request_count = len(repair.maintenance_request_ids)
+
+    def action_view_maintenance(self):
+        return {
+            'name': 'Child equipment of %s' % self.name,
+            'type': 'ir.actions.act_window',
+            'res_model': 'maintenance.request',
+            'res_id': self.id,
```
Comment (etobella) [inline maintenance_request_repair/models/repair_order.py]: This is probably wrong

File: maintenance_request_repair/models/repair_order.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        string='# Maintenances'
+    )
+
+    @api.depends('maintenance_request_ids')
+    def _compute_maintenance_request_count(self):
+        for repair in self:
+            repair.maintenance_request_count = len(repair.maintenance_request_ids)
+
+    def action_view_maintenance(self):
+        return {
```
Comment (etobella) [inline maintenance_request_repair/models/repair_order.py]: It is easier to use something like: 
- https://github.com/OCA/maintenance/blob/13.0/maintenance_project/models/project_project.py#L26

Also, you could try to show only the form if there is only one element, like:
- https://github.com/odoo/odoo/blob/13.0/addons/purchase_stock/models/purchase.py#L120

**Review by etobella — COMMENTED** [review]
Just a few comments.

File: maintenance_request_repair/models/repair_order.py (hunk 4 above)
Comment (marcelsavegnago) [reply maintenance_request_repair/models/repair_order.py]: Yes.. sorry.. :D

File: maintenance_request_repair/views/repair_order.xml (hunk 3 above)
Comment (marcelsavegnago) [reply maintenance_request_repair/views/repair_order.xml]: I'm thinking about this process and it makes sense to make it visible. I have a use case for repairing hospital equipment where this is useful.

File: maintenance_request_repair/models/repair_order.py (hunk 5 above)
Comment (marcelsavegnago) [reply maintenance_request_repair/models/repair_order.py]: Thanks

**Comment by marcelsavegnago** [comment]
@etobella @sergiocorato Could you please review this PR?

**Comment by marcelsavegnago** [comment]
@eduaparicio Could you please review this PR?

**Review by etobella — APPROVED** [review]

**Review by jpcarassato — APPROVED** [review]
Test OK!

**Comment by etobella** [comment]
@marcelsavegnago as Comments from @sergiocorato are already fulfilled and were cosmetic changes, I think we can proceed with /ocabot merge nobump

**Pull request merged**